MdeModulePkg/Application/UiApp/FrontPage.c

```diff
@@ -564,6 +571,291 @@ WarnNoBootableMedia (
+    Rsdp->Signature[5],
+    Rsdp->Signature[6],
+    Rsdp->Signature[7]
+  ));
+  if (CompareMem(Rsdp->Signature, RSDP_SIGNATURE, 8) != 0) {
+      DEBUG ((EFI_D_INFO, "    RSDP invalid signature\n"));
+      return NULL;
+  }
+  DEBUG ((EFI_D_INFO, "    Revision: 0x%x\n", Rsdp->Revision));
+
+  RsdtPtr = (UINTN)Rsdp->RsdtAddress;
+  DEBUG ((EFI_D_INFO, "  RSDT 0x%x\n", RsdpPtr));
+  Rsdt = (ACPI_SDT_HEADER*)RsdtPtr;
+  DEBUG ((EFI_D_INFO, "    Signature: '%c%c%c%c'\n",
+    Rsdt->Signature[0],
+    Rsdt->Signature[1],
+    Rsdt->Signature[2],
+    Rsdt->Signature[3]
+  ));
+  if (CompareMem(Rsdt->Signature, RSDT_SIGNATURE, 4) != 0) {
+      DEBUG ((EFI_D_INFO, "    RSDT invalid signature\n"));
+      return NULL;
+  }
+  DEBUG ((EFI_D_INFO, "    Revision: 0x%x\n", Rsdt->Revision));
+  DEBUG ((EFI_D_INFO, "    Length: 0x%x\n", Rsdt->Length));
+
+  for (Index = sizeof(ACPI_SDT_HEADER); Index < Rsdt->Length; Index += 4) {
+      TablePtr = (UINTN)(*(UINT32*)(RsdtPtr + Index));
+      DEBUG ((EFI_D_INFO, "  Table %d: 0x%x\n", Index, TablePtr));
+      Table = (ACPI_SDT_HEADER*)TablePtr;
+      DEBUG ((EFI_D_INFO, "    Signature: '%c%c%c%c'\n",
+        Table->Signature[0],
+        Table->Signature[1],
+        Table->Signature[2],
+        Table->Signature[3]
+      ));
+      DEBUG ((EFI_D_INFO, "    Revision: 0x%x\n", Table->Revision));
+      DEBUG ((EFI_D_INFO, "    Length: 0x%x\n", Table->Length));
+
+      if (CompareMem(Table->Signature, Name, 4) == 0) {
+          DEBUG ((EFI_D_INFO, "  Match found\n"));
+          return Table;
+      }
+  }
+
+  DEBUG ((EFI_D_INFO, "  No match found\n"));
+  return NULL;
+}
+
+STATIC VOID FirmwareConfigurationInformation(VOID) {
+    EFI_STRING_ID Token;
+
+    Token = STRING_TOKEN (STR_VIRTUALIZATION);
+    if (StandardSignatureIsGenuineIntel()) {
+        HiiSetString (gFrontPagePrivate.HiiHandle, Token, L"Intel Virtualization", NULL);
+
+        Token = STRING_TOKEN (STR_VIRTUALIZATION_STATUS);
+        CPUID_VERSION_INFO_ECX VersionInfoEcx;
+        AsmCpuid (CPUID_VERSION_INFO, NULL, NULL, &VersionInfoEcx.Uint32, NULL);
+        if (VersionInfoEcx.Bits.VMX) {
+            HiiSetString (gFrontPagePrivate.HiiHandle, Token, L"VT-x: Active", NULL);
+        } else {
+            HiiSetString (gFrontPagePrivate.HiiHandle, Token, L"VT-x: Deactivated", NULL);
+        }
+
+        Token = STRING_TOKEN (STR_IOMMU_STATUS);
+        CHAR8 TableName[4] = {'D', 'M', 'A', 'R'};
+        if (FindAcpiTable(TableName)) {
+            HiiSetString (gFrontPagePrivate.HiiHandle, Token, L"VT-d: Active", NULL);
+        } else {
+            HiiSetString (gFrontPagePrivate.HiiHandle, Token, L"VT-d: Deactivated", NULL);
+        }
+
+        Token = STRING_TOKEN(STR_ME_STATUS);
+        //TODO: proper test for ME
+        BOOLEAN me_active = FALSE;
+        if (!me_active) {
+            HiiSetString (gFrontPagePrivate.HiiHandle, Token, L"The Intel Management Engine is disabled at runtime to increase security.", NULL);

```

Comment:
Does the justification "to increase security" need to be here?

(I feel like we should have a more comprehensive statement somewhere about our position on, and handling of, the IME that we can refer users to.)

Reply:
Good idea, have an example of what information you'd want to add?

Reply:
A general overview:

- A brief about what the IME is
- Link to Wikipedia for more information about IME
- Why we disable it
  - Proprietary OS running on CPU with known vulnerabilities in its modules
  - Features generally are not useful to our users and/or not used
  - Call out IBG specifically as hostile to user modification?

And a more technical explanation:

- That the IME is required for initialization (since Nehalem?), so it cannot be completely removed
- `me_cleaner` does not currently support the versions of the IME used on our systems
- So, instead, we use the HECI command to disable runtime components during early boot
  - Maybe also cover HAP mode for earlier models

Reply:
I think we can address these in a future PR, if that is okay with you